app/main/request.ts

```diff
@@ -0,0 +1,148 @@
+import {app, net} from 'electron';
+import fs from 'fs';
+import path from 'path';
+import * as Messages from '../resources/messages';
+import Logger from '../renderer/js/utils/logger-util';
+import {ServerConf} from '../renderer/js/utils/domain-util';
+import ReconnectUtil from '../renderer/js/utils/reconnect-util';
+
+/* Request: domain-util */
+
+const defaultIconUrl = '../renderer/img/icon.png';
+
+const logger = new Logger({
+	file: 'domain-util.log',
+	timestamp: true
+});
+
+const generateFilePath = (url: string): string => {
+	const dir = `${app.getPath('userData')}/server-icons`;
+	const extension = path.extname(url).split('?')[0];
+
+	let hash = 5381;
+	let {length} = url;
+
+	while (length) {
+		hash = (hash * 33) ^ url.charCodeAt(--length);
+	}
+
+	// Create 'server-icons' directory if not existed
+	if (!fs.existsSync(dir)) {
+		fs.mkdirSync(dir);
+	}
+
+	return `${dir}/${hash >>> 0}${extension}`;
+};
+
+export const _getServerSettings = async (domain: string, Session: Electron.session): Promise<ServerConf> => {
+	const serverSettingsOptions: Electron.ClientRequestConstructorOptions = {
+		url: domain + '/api/v1/server_settings',
+		session: Session
+	};
+
+	return new Promise((resolve, reject) => {
+		const request = net.request(serverSettingsOptions);
+		let data = '';
+
+		request.on('response', response => {
+			response.on('data', chunk => {
+				data += chunk;
+			});
+
+			response.on('end', () => {
+				if (response.statusCode === 200) {
+					const {realm_name, realm_uri, realm_icon} = JSON.parse(data);
+					if (
+						typeof realm_name === 'string' &&
+						typeof realm_uri === 'string' &&
+						typeof realm_icon === 'string'
+					) {
+						resolve({
+							// Some Zulip Servers use absolute URL for server icon whereas others use relative URL
+							// Following check handles both the cases
+							icon: realm_icon.startsWith('/') ? realm_uri + realm_icon : realm_icon,
+							url: realm_uri,
+							alias: escape(realm_name)
```

Comment:
Need to import `import escape from 'escape-html';` .
Currently the PR uses built-in `escape`, which's behavior is different from what we have in `domain-util.ts` .